StackExchange.Redis/StackExchange/Redis/ConnectionMultiplexer.cs

```diff
@@ -37,13 +37,16 @@ public static Task<T> ObserveErrors<T>(this Task<T> task)
             return task;
         }
 
+        [ThreadStatic]
+        public static bool ContinueOnCapturedContext;
```

Comment:
I think what this PR aims to do is great, but I'm dubious about this as the chosen implementation; it seems... unusual, to me - it seems that this should be something we can drill down into the appropriate methods, rather than relying on thread-static variables, which are dangerous on a good day, and downright confusing when `async` and the TPL are involved. Any specific reason you preferred this option?